src/pages/grants.tsx

```diff
@@ -1,23 +1,48 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 
+import { useLocation } from '@gatsbyjs/reach-router'
 import Head from 'decentraland-gatsby/dist/components/Head/Head'
 import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
 import { Container } from 'decentraland-ui/dist/components/Container/Container'
+import { NotMobile } from 'decentraland-ui/dist/components/Media/Media'
 import isEmpty from 'lodash/isEmpty'
+import toSnakeCase from 'lodash/snakeCase'
+import Grid from 'semantic-ui-react/dist/commonjs/collections/Grid/Grid'
 
+import CurrentGrantsBanner from '../components/Grants/Current/CurrentGrantsBanner'
 import CurrentGrantsList from '../components/Grants/Current/CurrentGrantsList'
-import PastGrantsList from '../components/Grants/Past/PastGrantsList'
+import RequestBanner from '../components/Grants/RequestBanner'
 import BurgerMenuLayout from '../components/Layout/BurgerMenu/BurgerMenuLayout'
 import LoadingView from '../components/Layout/LoadingView'
 import MaintenanceLayout from '../components/Layout/MaintenanceLayout'
 import Navigation, { NavigationTab } from '../components/Layout/Navigation'
+import CategoryFilter from '../components/Search/CategoryFilter'
+import StatusFilter from '../components/Search/StatusFilter'
+import { GrantStatus, NewGrantCategory, OldGrantCategory } from '../entities/Grant/types'
+import { toGrantStatus, toProposalGrantCategory } from '../entities/Grant/utils'
+import { GrantWithUpdateAttributes } from '../entities/Proposal/types'
 import useGrants from '../hooks/useGrants'
 import { isUnderMaintenance } from '../modules/maintenance'
 
+function filterDisplayableGrants(grants: GrantWithUpdateAttributes[], type: string | null, status: string | null) {
```

Comment:
I think we could write this like this to avoid going through all grants twice:

```typescript
function filterDisplayableGrants(grants: GrantWithUpdateAttributes[], type: string | null, status: string | null) {
  return status || type
    ? grants.filter(
        (grant) =>
          (type ? toSnakeCase(grant.configuration.category) === type : true) &&
          (status ? toSnakeCase(grant.status) === status : true)
      )
    : grants
}
```